### PR TITLE
Make carbonifer available as Go library

### PR DESCRIPTION
**Pull request by Joffref**
Closes: #40

Thanks to this PR `GetEstimationFromInstanceType()`, `GetEstimation()` and  `GetResource()` will be available as part of carbonifer Go Lib.

Note to the maintainers: I had struggle with the way viper is used to import data, I make an ugly workaround. I'm open to your suggestion on this point in order to find the best way to patch this. :smile:

### CODE REVIEW

**Comment by obierlaire** [comment]
Thanks a lot @Joffref for this first contribution ❤️ That's much appreciated !
I haven't had time to properly review it yet, that's a nice work!
However, I've noticed a few things: 
- CI is failing: [golangci-cli  found one tiny error ](https://github.com/carboniferio/carbonifer/actions/runs/4257086100/jobs/7422172534)
- Even if it's pretty straightforward `GetEstimation()` should have a test. It's not a big deal if you don't have time though.

The viper part does not look that ugly, I'll have a look later to see if we can find a nicer way

Again, thanks!

**Comment by Joffref** [comment]
Hey @obierlaire, I've made few changes on the PR, based on your comments. Hope it will be good this time, let me know if there is something to modify in it.

**Review by obierlaire — CHANGES_REQUESTED** [review]
I think you need to rebase once https://github.com/carboniferio/carbonifer/pull/47 will be merged

**Comment by Joffref** [comment]
> I think you need to rebase once #47 will be merged

I agree :+1: Waiting to the update on your side

**Comment by obierlaire** [comment]
#47 merged, you can rebase your branch

File: pkg/estimate/estimate_test.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+						"Haswell",
+						"AMD EPYC Rome",
+						"AMD EPYC Milan",
+					},
+					MemoryMb:          8192,
+					VCPUs:             2,
+					Storage:           resources.Storage{},
+					ReplicationFactor: 0,
+				},
+				Power:           decimal.NewFromFloat(4.8677297799),
```
[the remark this answers is not in the document]
Comment (Joffref) [reply pkg/estimate/estimate_test.go:58]: Yes, after a retry with default config it appears that those values are wrong.

File: pkg/estimate/estimate.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+func toInternalComputeResource(resource resources.GenericResource) internalResources.ComputeResource {
+	return internalResources.ComputeResource{
+		Identification: resource.GetIdentification(),
+		Specs: &internalResources.ComputeResourceSpecs{
+			GpuTypes:          resource.GPUTypes,
+			HddStorage:        resource.Storage.HddStorage,
+			SsdStorage:        resource.Storage.SsdStorage,
+			MemoryMb:          resource.MemoryMb,
+			VCPUs:             resource.VCPUs,
+			CPUType:           resource.CPUTypes[0], // TODO: Support multiple CPU types
```
[the remark this answers is not in the document]
Comment (Joffref) [reply pkg/estimate/estimate.go:55]: Make sense

**Comment by obierlaire** [comment]
@Joffref I pushed [ad7d82f](https://github.com/carboniferio/carbonifer/pull/41/commits/ad7d82f16f162d2b19aaf27fba748a5099f8b19e) to your PR in order to have one single rationalized way to init Viper and set `data.path` variable, always from the go project root directory, and not relative to the caller. That's more DRY and seems to work...

**Review by obierlaire — APPROVED** [review]

**Comment by Joffref** [comment]
Nice one ! It will be better like this :+1:

**Pull request merged**